pkg/cmd/server/kubernetes/master_config.go

```diff
@@ -205,39 +217,48 @@ func BuildKubernetesMasterConfig(options configapi.MasterConfig, requestContextM
 		proxyClientCerts = append(proxyClientCerts, clientCert)
 	}
 
-	// TODO you have to know every APIGroup you're enabling or upstream will panic.  Its alternative to panicing is Fataling
-	// It needs a refactor to return errors
-	storageDestinations := genericapiserver.NewStorageDestinations()
-	// storageVersions is a map from API group to allowed versions that must be a version exposed by the REST API or it breaks.
-	// We need to fix the upstream to stop using the storage version as a preferred api version.
-	storageVersions := map[string]string{}
-
-	enabledKubeVersions := configapi.GetEnabledAPIVersionsForGroup(*options.KubernetesMasterConfig, configapi.APIGroupKube)
-	if len(enabledKubeVersions) > 0 {
-		kubeStorageVersion := unversioned.GroupVersion{Group: configapi.APIGroupKube, Version: options.EtcdStorageConfig.KubernetesStorageVersion}
-		databaseStorage, err := NewEtcdStorage(etcdClient, kubeStorageVersion, options.EtcdStorageConfig.KubernetesStoragePrefix)
-		if err != nil {
-			return nil, fmt.Errorf("Error setting up Kubernetes server storage: %v", err)
-		}
-		storageDestinations.AddAPIGroup(configapi.APIGroupKube, databaseStorage)
-		storageVersions[configapi.APIGroupKube] = options.EtcdStorageConfig.KubernetesStorageVersion
+	resourceEncodingConfig := genericapiserver.NewDefaultResourceEncodingConfig()
+	resourceEncodingConfig.SetVersionEncoding(
+		configapi.APIGroupKube,
+		unversioned.GroupVersion{Group: configapi.APIGroupKube, Version: options.EtcdStorageConfig.KubernetesStorageVersion},
+		configapi.SchemeGroupVersion,
```

Comment:
`configapi.SchemeGroupVersion` is the groupversion of our config objects... I'd expect something like this instead:
```
	resourceEncodingConfig.SetVersionEncoding(
		kapi.GroupName,
		unversioned.GroupVersion{Group: kapi.GroupName, Version: options.EtcdStorageConfig.KubernetesStorageVersion},
		kapi.SchemeGroupVersion,
	)
```